### PR TITLE
feat: ASAP-288 Add middle name and nickname to user models

### PR DESCRIPTION
**Pull request by peterstarling**

### CODE REVIEW

**Review by lctrt — CHANGES_REQUESTED** [review]
Would be great to remove duplication of the displayName function :) Otherwise looks great!

**Review by lctrt — APPROVED** [review]

File: apps/crn-server/test/routes/user.route.test.ts [hunk 1]
```diff
@@ -379,70 +406,59 @@ describe('/users/ route', () => {
...
+            expect(response.status).toBe(200);
+          });
+
+          test('Should be able to provide an empty string for the %s parameter ', async () => {
+            const response = await supertest(appWithMockedAuth)
+              .patch(`/users/${userId}`)
+              .send({ [parameter]: '' });
+
+            expect(response.status).toBe(200);
+          });
```
Comment (gabiayako) [inline apps/crn-server/test/routes/user.route.test.ts:449]: I believe these two are the same

File: apps/crn-server/test/data-providers/contentful/users.data-provider.test.ts [hunk 2]
```diff
@@ -97,6 +97,21 @@ describe('User data provider', () => {
       expect(result).toEqual(null);
     });
 
+    test.each(['middleName', 'nickname'] satisfies Array<keyof UserDataObject>)(
+      'Should not return a field when its an emptry string for %s',
```
Comment (gabiayako) [inline apps/crn-server/test/data-providers/contentful/users.data-provider.test.ts:101]:
```suggestion
      'Should not return a field when its an empty string for %s',
```

File: apps/gp2-server/test/data-providers/user.data-provider.test.ts [hunk 3]
```diff
@@ -62,6 +62,23 @@ describe('User data provider', () => {
       expect(result).toEqual(getUserDataObject());
     });
 
+    test.each(['middleName', 'nickname'] satisfies Array<
+      keyof gp2Model.UserDataObject
+    >)(
+      'Should not return a field when its an emptry string for %s',
```
Comment (gabiayako) [inline apps/gp2-server/test/data-providers/user.data-provider.test.ts:68]:
```suggestion
      'Should not return a field when its an empty string for %s',
```

**Pull request merged**